starter-code/js/index.js

```diff
@@ -1,52 +1,76 @@
+
+//   calculatePriceButton.onclick = getTotalPrice;
+//   createItemButton.onclick = createNewItem;
+
+//   for(var i = 0; i<deleteButtons.length ; i++){
+//     deleteButtons[i].onclick = deleteItem;
+//   }
+// };
+
+var priceString = document.querySelectorAll(".price");
+var inputValue = document.querySelectorAll(".input");
+var displayTotalPrice = document.querySelectorAll(".total-price");
+var calculateBtn = document.querySelector(".btn-success");
+var deleteBtn = document.querySelectorAll(".btn-delete");
+var createBtn = document.querySelector(".btn-create");
+var createProduct = document.querySelector(".new-creation1");
+var createProduct1 = document.querySelector(".new-creation2");
+var addList = document.querySelector(".shopping-cart");
+
+calculateBtn.onclick = doCalculate;
+
+//calculate the prices + quantity
+function getTotalPrice(){
+   for ( i = 0; i < priceString.length; i++){
+    displayTotalPrice[i].innerHTML = (priceString[i].innerHTML) * (inputValue[i].value);
+  };
+};
 
-window.onload = function(){
-  var calculatePriceButton = document.getElementById('calc-prices-button');
-  var createItemButton = document.getElementById('new-item-create');
-  var deleteButtons = document.getElementsByClassName('btn-delete');
+var finalResult = 0;
 
-  calculatePriceButton.onclick = getTotalPrice;
-  createItemButton.onclick = createNewItem;
+// calculate the finalprice
+function finalTotalPrice(){
+  var bigTotal = document.querySelector(".final-result");
+  for (j = 0; j < displayTotalPrice.length; j++){
+    finalResult += displayTotalPrice[j].innerHTML;
+    bigTotal.innerHTML = Number(finalResult);
+  };
+};
 
-  for(var i = 0; i<deleteButtons.length ; i++){
-    deleteButtons[i].onclick = deleteItem;
-  }
+//did the calculation
+function doCalculate (){
+  getTotalPrice();
+  finalTotalPrice();
 };
+
+// append the click delete functionality to the buttons that are there at first
+deleteBtn.forEach(function (oneButton) {
+  oneButton.onclick = function () {
+      var isOkay = confirm("Are you sure?");
+      if (isOkay) {
+        oneButton.parentNode.remove();
+      }
+  };
+});
+
+//create new item (but not working)
+createBtn.onclick = function(){
+  var addProduct = document.createElement("div");
+  addProduct.setAttribute("class", ".product" );
+  addProduct.innerHTML = createProduct.value +  createProduct1.value + '<label class="quatity">Qty</label>' +' <input class="input" type="number" placeholder="0" />' + ' $<span class="total-price"></span>' + ' <button class="btn .btn-delete">Delete</button>';
+  document.body.appendChild(addProduct);
+    var newBtn = addProduct.querySelector(".btn-delete");
+    newBtn.onclick = function() {
+        //confirm with the user that the delete is okay
+        var isOkay = confirm("Are you sure?");
+        if (isOkay) {
+          oneButton.parentNode.remove();
+        };
+    };
+    createProduct.value="";
+    createProduct1.value="";
+};
```

Comment:
You are trying to mix many many steps at once in this function and a few features are not working properly : the delete function is not hooked up to the button, and also the price calculation doesn't take into account the newly created products.

If only for debugging purposes, I would advise you to separate every piece of logic inside different functions : you can extract the delete item function for exemple.  

Here, the problem is that the interpretor can't parse the html right after is has been hardcoded in innerHTML, the clean way to create elements in the dom is to use the proper functions : createElement, textContent, appendChild etc. 
Here is an exemple for extraction the creation of the delete button in a different function, but you can do the same with the label, input and span: 
```js
createBtn.onclick = function() {
  var addProduct = document.createElement("div");
  addProduct.setAttribute("class", ".product");
  addProduct.innerHTML =
    createProduct.value +
    createProduct1.value +
    '<label class="quatity">Qty</label>' +
    ' <input class="input" type="number" placeholder="0" />' +
    ' $<span class="total-price"></span>';

  document.body.appendChild(addProduct);
  var newBtn = createDeleteButton();
  addProduct.appendChild(newBtn);
  createProduct.value = "";
  createProduct1.value = "";
};

function createDeleteButton() {
  var div = document.createElement("div");
  div.className = "item-delete col-xs-3";
  var buttonNode = document.createElement("button");

  buttonNode.className = "btn btn-delete";
  buttonNode.innerHTML = "Delete";
  buttonNode.onclick = function() {
    var isOkay = confirm("Are you sure?");
    if (isOkay) {
      buttonNode.parentNode.parentNode.remove();
    }
  };
  div.appendChild(buttonNode);

  return div;
}
```
Here is a more detailed correction : https://github.com/ironhack-paris-webdev-october-2018/lab-dom-ironhack-cart/blob/master/solution-code/js/index.js